server/data/strapiApiTypes.ts

```diff
@@ -11,14 +11,22 @@ export type ListResponse<T> = {
     }
   }
 }
+type HasComponents = { components: { data: ComponentListResponseDataItem[] } }
+type HasTeam = { team: TeamResponse }
+type HasServiceArea = { service_area: ServiceAreaResponse }
+type HasProduct = { attributes: { product?: { data: { attributes: ProductWithTeamAndServiceArea } } } }
 
-export type Product = components['schemas']['Product']
+type ProductWithTeamAndServiceArea = Omit<Omit<ProductResponse['data']['attributes'], 'team'>, 'serviceArea'> &
+  HasTeam &
+  HasServiceArea
+
+export type Product = Omit<components['schemas']['Product'], 'components'> & HasComponents
```

Comment:
This is all very mind melty but the idea was to patch the types rather than rely on manually editing the massive types file